File: MATTHEW/leetcode/AIcoding/5.Qwen-3/cpp_solutions/3151-special-array-i.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int specialArray(vector<int>& nums) {
        int count = 0;
        for (int i = 0; i <= 1000; ++i) {
            int temp = 0;
            for (int num : nums) {
                if (num >= i) {
                    temp++;
                }
            }
            if (temp == i) {
                return i;
            }
        }
        return -1;
    }
};

File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/rust_solutions/3151-special-array-i.rs
impl Solution {
    pub fn special_array(nums: Vec<i32>) -> i32 {
        for i in 0..=nums.len() as i32 {
            let mut count = 0;
            for &num in &nums {
                if num >= i {
                    count += 1;
                }
            }
            if count == i {
                return i;
            }
        }
        -1
    }
}